src/main/java/io/cryostat/expressions/MatchExpressions.java

```diff
@@ -63,7 +65,34 @@ public V2Response test(RequestData requestData) throws ScriptException {
                                         .ifPresent(targets::add));
         var matched =
                 targetMatcher.match(new MatchExpression(requestData.matchExpression), targets);
-        return V2Response.json(Response.Status.OK, matched);
+        // Convert matched expression to JSON object directly
+        JsonObject jsonResponse = createMatchedResponse(matched);
+
+        return Response.ok(jsonResponse.encode(), MediaType.APPLICATION_JSON).build();
+    }
+
+    private JsonObject createMatchedResponse(MatchExpression.MatchedExpression matched) {
+        JsonArray targetsJson = new JsonArray();
+        matched.targets()
+                .forEach(
+                        target -> {
+                            JsonObject targetJson = new JsonObject();
+                            targetJson.put("connectUrl", target.connectUrl);
+                            targetJson.put("alias", target.alias);
+                            targetsJson.add(targetJson);
+                        });
+
+        return new JsonObject()
```

Comment:
Here again, this is just manually creating the equivalent to a V2Response, which is mainly what we're trying to avoid doing anymore. I think the response body here would just be the `MatchedExpression` object.